awx/main/access.py

```diff
@@ -2092,11 +2092,23 @@ def can_add(self, data):
         if not data:  # So the browseable API will work
             return Organization.accessible_objects(self.user, 'workflow_admin_role').exists()
 
-        return bool(
-            self.check_related('organization', Organization, data, role_field='workflow_admin_role', mandatory=True)
-            and self.check_related('inventory', Inventory, data, role_field='use_role')
-            and self.check_related('execution_environment', ExecutionEnvironment, data, role_field='read_role')
-        )
+        if not self.check_related('organization', Organization, data, role_field='workflow_admin_role', mandatory=True):
+            if data.get('organization', None) is None:
+                self.messages['workflow_job_template'] = [_('An organization is required to create a workflow job template for normal user')]
+                return False
+
+            self.messages['workflow_job_template'] = [_('You do not have workflow_admin_role to the organization')]
+            return False
+
+        if not self.check_related('inventory', Inventory, data, role_field='use_role'):
+            self.messages['workflow_job_template'] = [_('You do not have use_role to the inventory')]
+            return False
+
+        if not self.check_related('execution_environment', ExecutionEnvironment, data, role_field='read_role'):
+            self.messages['workflow_job_template'] = [_('You do not have read_role to the execution environment')]
```

Comment:
These 2 messages would be fine, but I'd argue the keys (which become keys in the response) should be "inventory" and then "execution_environment"